### ISSUE TITLE
Call Nodes

### ISSUE THREAD

**Issue by pasetti** [pnp-software/fcp-editor#6]
A "Call Node" is a type of action node in an FCP (see also issue #5 ). It describes the fact that the FCP is calling another FCP. A Call Node is defined by the following properties in its "Action Node Properties" panel:

- The action node identifier (same as in action nodes)
- The action description (same as in action nodes)
- The name of the FCP which is being called (the "Target FCP")
- The values of the inputs of the FCP being called
- The mode of the call

NB: Action nodes have the entryType property ("Call Function", "Define Function" or "Run Code"). This property is not needed for Call Nodes.

The **Name of the FCP** which is being called should be selected from a pull-down menu which presents to the user the names and titles of all procedures (both those in the current project and those in other projects). To illustrate, suppose that we have two FCPs: FCP_A and FCP_B and suppose that node N1 of FCP_A is a Call Node which calls FCP_B. In the json representation, node N1 should have an attribute called 'targetNode" which points to FCP_B.  

For the **values of the inputs** of the target FCP, the "Action Node Properties" panel should offer a list of all input parameters of the selected target FCP and the user should be asked to fill in appropriate values. Here, too, we must decide what to do when the user has selected a certain input parameter and has assigned a value to it but then the user deletes the input parameter or changes its name. A similar situation arises when the user assigns the values of the input parameters of a target FCP but then, later, a new input parameter is defined for the target FCP.

The **mode of the call** is a flag which can have only two values: "RUN" (to indicate that, after starting the target FCP, the parent FCP must wait until it has terminated) and "RELEASE" (to indicate that, after starting the target FCP, the parent FCP may immediately continue execution). The RELEASE situation is one where the parent FCP and the target FCP run in parallel.
